tests/acceptance/blocks/verticalresults.js

```diff
@@ -66,6 +68,42 @@ class VerticalResults {
     return this._selector.exists;
   }
 
+  /**
+   * Wait for results to load on page by checking query response status and searchComplete state
+   * (timeout is set to 10 seconds)
+   */
+  async waitOnSearchComplete() {
+    const responseWaitTimeout = 10000;
+    const waitTimeInterval = 200;
+    let totalWaitTime = 0;
+    while (totalWaitTime < responseWaitTimeout 
+      && this._queryRequestLogger.requests.find(r => !r.response || r.response.statusCode !== 200)) {
```

Comment:
Is this code waiting while an invalid response exists? is there a reason for that approach rather than waiting until a successful response has been logged?

Reply:
Logger will hold all request/response throughout a test so just checking for a successful response might accidentally check on previous requests. But, after some thoughts, I updated this method check for a successful response in the loop, and clear out the logger at the end of this method. This make sure we are always verifying the latest response.